ipatests/test_webui/test_selfservice.py

```diff
@@ -49,4 +60,231 @@ def test_crud(self):
+    def test_del_multiple_permission(self):
+        """
+        Try to delete multiple self service permission
+        """
+        self.init_app()
+        self.add_record(data_selfservice.ENTITY, data_selfservice.DATA1)
+        self.add_record(data_selfservice.ENTITY, data_selfservice.DATA)
+        self.delete(data_selfservice.ENTITY,
+                    [data_selfservice.DATA1,
+                     data_selfservice.DATA])
+        # check if record deleted
+        for key in [data_selfservice.DATA1['pkey'],
+                    data_selfservice.DATA['pkey']]:
+            self.assert_record(key, negative=True)
+
+    @screenshot
+    def test_permission_using_enter_key(self):
+        """
+        Try to add/delete persmission using enter key
+        """
+        # try to add using enter key
+        self.init_app()
+        self.add_record(data_selfservice.ENTITY,
+                        data_selfservice.DATA1,
+                        dialog_btn=None)
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait()
+        self.assert_notification(assert_text=SERVICE_ADDED)
+        self.assert_record(data_selfservice.DATA1['pkey'])
+        self.close_notifications()
+
+        # try to delete using enter key
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.select_record(data_selfservice.DATA1['pkey'])
+        self.facet_button_click('remove')
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait()
+        self.assert_notification(assert_text='1 item(s) deleted')
+        self.close_notifications()
+
+    @screenshot
+    def test_reset_sshkey_permsission(self):
+        """
+        Try to delete sshkey after altering sshkey permission
+        """
+        pwd = self.config.get('ipa_password')
+
+        self.init_app()
+        self.add_record(user.ENTITY, user.DATA, navigate=False)
+        reset_passwd(self, user.PKEY, pwd)
+        self.logout()
+        self.login(user.PKEY, password=pwd, new_password=pwd)
+        self.add_sshkey_to_record(user.SSH_RSA, user.PKEY, navigate=True)
+        self.assert_num_ssh_keys(1)
+        self.close_notifications()
+        self.logout()
+
+        self.init_app()
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.navigate_to_record('Users can manage their own SSH public keys')
+        # to pass the validator, check some option from the options
+        self.check_option('attrs', 'carlicense')
+        # uncheck the ipasshpubkey checkbox
+        self.check_option('attrs', 'ipasshpubkey')
+        self.facet_button_click('save')
+        self.close_notifications()
+        self.logout()
+
+        # check if delete button is visible for ssh key
+        self.login(user.PKEY, password=pwd)
+        s = "button[name=remove]"
+        facet = self.get_facet()
+        btn = self.find(s, By.CSS_SELECTOR, facet, strict=True)
```

Comment:
This test assumes that there is no other widget with 'remove' button visible. E.g. if there would be any editable multivalued widget(e.g. telephone number) with some value set then this test would incorrectly fail as it could pick the 'remove' button from the other widget. Right now it is not a problem but might create hard to debug issue in the future. IMO the selector `s` should be limited to `ipasshpubkey` widget.